rinja_parser/src/expr.rs

```diff
@@ -190,26 +191,40 @@ impl<'a> Expr<'a> {
     expr_prec_layer!(band, shifts, token_bitand);
     expr_prec_layer!(shifts, addsub, alt((tag(">>"), tag("<<"))));
     expr_prec_layer!(addsub, muldivmod, alt((tag("+"), tag("-"))));
-    expr_prec_layer!(muldivmod, is_defined, alt((tag("*"), tag("/"), tag("%"))));
+    expr_prec_layer!(muldivmod, is_as, alt((tag("*"), tag("/"), tag("%"))));
 
-    fn is_defined(i: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
+    fn is_as(i: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
         let start = i;
         let (i, lhs) = Self::filtered(i, level)?;
-        let (i, rhs) = opt(preceded(
-            ws(keyword("is")),
-            opt(terminated(opt(keyword("not")), ws(keyword("defined")))),
-        ))(i)?;
-        let is_neg = match rhs {
-            None => return Ok((i, lhs)),
-            Some(None) => {
+        let (j, rhs) = opt(ws(identifier))(i)?;
+        let i = match rhs {
+            Some("is") => j,
+            Some("as") => {
+                let (i, target) = opt(identifier)(j)?;
+                let target = target.unwrap_or_default();
+                if crate::PRIMITIVE_TYPES.contains(&target) {
+                    return Ok((i, WithSpan::new(Self::As(Box::new(lhs), target), start)));
+                } else {
+                    return Err(nom::Err::Failure(ErrorContext::new(
+                        "operator `as` expects the name of a primitive type on its right-hand side",
```

Comment:
Sorry, completely missed this message error typo:

```suggestion
                        "`as` operator expects the name of a primitive type on its right-hand side",
```

I would also add the invalid rhs used like:

```suggestion
                        "`as` operator expects the name of a primitive type on its right-hand side, found `{target}`",
```